src/examples/password_encryption.cpp

```diff
@@ -0,0 +1,140 @@
+#include <botan/aead.h>
+#include <botan/assert.h>
+#include <botan/auto_rng.h>
+#include <botan/hex.h>
+#include <botan/pwdhash.h>
+
+#include <iostream>
+
+namespace {
+
+template <typename OutT = std::vector<uint8_t>, typename... Ts>
+OutT concat(Ts&&... buffers) {
+   OutT out;
+   out.reserve((buffers.size() + ... + 0));
+   (out.insert(out.end(), buffers.begin(), buffers.end()), ...);
+   return out;
+}
+
+template <typename Out, typename In>
+Out as(const In& data) {
+   return Out(data.data(), data.data() + data.size());
+}
+
+constexpr size_t salt_length = 16;
+
+Botan::secure_vector<uint8_t> derive_key_material(std::string_view password,
+                                                  std::span<const uint8_t> salt,
+                                                  size_t output_length) {
+   // Here, we use statically defined password hash parameters. Alternatively
+   // you could use Botan::PasswordHashFamily::tune() to automatically select
+   // parameters based on your desired runtime and memory usage.
+   //
+   // Defining those parameters highly depends on your use case and the
+   // available compute and memory resources on your target platform.
+   const std::string pbkdf_algo = "Argon2id";
+   constexpr size_t M = 256 * 1024;  // kiB
+   constexpr size_t t = 4;           // iterations
+   constexpr size_t p = 2;           // parallelism
+
+   auto pbkdf = Botan::PasswordHashFamily::create_or_throw(pbkdf_algo)->from_params(M, t, p);
+   BOTAN_ASSERT_NONNULL(pbkdf);
+
+   Botan::secure_vector<uint8_t> key(output_length);
+   pbkdf->hash(key, password, salt);
+
+   return key;
+}
+
+std::unique_ptr<Botan::AEAD_Mode> prepare_aead(std::string_view password,
+                                               std::span<const uint8_t> salt,
+                                               Botan::Cipher_Dir direction) {
+   auto aead = Botan::AEAD_Mode::create_or_throw("AES-256/GCM", direction);
+
+   // Stretch the password into enough cryptographically strong key material
+   // to initialize the AEAD with a key and nonce (aka. initialization vector).
+   const auto keydata_needed = aead->key_spec().minimum_keylength() + aead->default_nonce_length();
+   const auto keydata = derive_key_material(password, salt, keydata_needed);
+   const auto key = std::span{keydata}.first(aead->key_spec().minimum_keylength());
+   const auto nonce = std::span{keydata}.last(aead->default_nonce_length());
```

Comment:
Grabbing the keylength and nonce lengths into local variables would avoid a bit of repetition, save on a couple of virtual calls, and (perhaps marginally) improve readability.